src/test/components/UploadedRecordingsHome.test.js

```diff
@@ -145,4 +147,58 @@ describe('UploadedRecordingsHome', () => {
 
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  it('can rename stored profiles', async () => {
+    // Store a single profile that we will rename. The data inside is mostly
+    // arbirtrary except the "name" field.
+    // mockDate('4 Jul 2020 15:00');
+    await storeProfileData({
+      profileToken: 'FAKE_1234',
+      jwtToken: null,
+      publishedDate: new Date('4 Jul 2020 13:00'), // This is the future!
+      name: 'Initial Profile Name',
+      preset: null,
+      originHostname: 'https://mozilla.org',
+      meta: {
+        product: 'Firefox',
+        platform: 'Macintosh',
+        toolkit: 'cocoa',
+        misc: 'rv:62.0',
+        oscpu: 'Intel Mac OS X 10.12',
+      },
+      urlPath: '/public/FAKE_1234/marker-chart/',
+      publishedRange: { start: 2000, end: 40000 },
+    });
+    const store = blankStore();
+    {
+      const { findByText } = render(
+        <Provider store={store}>
+          <UploadedRecordingsHome />
+        </Provider>
+      );
+
+      expect(await findByText(/Initial Profile Name/)).toBeTruthy();
+    }
+
+    const newUrlState = stateFromLocation({
+      pathname: `/public/FAKE_1234/marker-chart/`,
+      search: '',
+      hash: '',
+    });
+    store.dispatch({
+      type: 'UPDATE_URL_STATE',
+      newUrlState,
+    });
```

Comment:
I'm just wondering why that part is needed :-) can you please explain this? If that's not needed just remove it please!

Reply:
Suggestion: maybe that's because you're testing `UploadedRecordingsHome`, maybe you should test `ListOfPublishedProfiles` instead?

Reply:
Since we are using `const hash = getHash(getState());` in the changeProfileName action, It is giving hash as null untill we do above UPDATE_URL_STATE action